docs/tutorials/mastering.md

```diff
@@ -0,0 +1,618 @@
+
+```console
+sudo snap install microk8s --channel=1.27-strict/stable
+sudo microk8s enable hostpath-storage
+sudo microk8s enable metallb:10.201.0.50-10.201.0.51
+sudo usermod -a -G snap_microk8s $USER
+newgrp snap_microk8s
+```
+
+Install Juju and bootstrap the controller to the local Microk8s install as a load balancer service. This will expose the Juju controller on the first allocated MetalLB address.
+
+```console
+mkdir -p ~/.local/share/juju
+sudo snap install juju --channel=3.1/stable
+juju bootstrap microk8s --config controller-service-type=loadbalancer sdcore
+```
+
+At this point, the Juju controller is ready to start managing external clouds. Add the Kubernetes clusters representing the user plane, control plane, and gNB simulator to Juju. This is done by using the Kubernetes configuration file generated when setting up the clusters above.
+
+```console
+export KUBECONFIG=control-plane-cluster.yaml
+juju add-k8s control-plane-cluster --controller sdcore
+export KUBECONFIG=user-plane-cluster.yaml
+juju add-k8s user-plane-cluster --controller sdcore
+export KUBECONFIG=gnb-cluster.yaml
+juju add-k8s gnb-cluster --controller sdcore
+```
+
+You may now proceed to deploy the SD-Core Control Plane and SD-Core User Plane
+
+## 2. Deploy SD-Core Control Plane
+
+The following steps build on the Juju controller which was bootstrapped and knows how to manage the SD-Core Control Plane Kubernetes cluster.
+
+First, create a Juju overlay file that specifies the Access and Mobility Management Function (AMF) host name and IP address for sharing with the radios. This host name must be resolvable by the gNB and the IP address must be reachable and resolve to the AMF unit. In the bootstrap step, we set the Control Plane MetalLB range to start at `10.201.0.100`, so that is what we use in the configuration.
+
+```console
+cat << EOF > control-plane-overlay.yaml
+applications:
+  amf:
+    options:
+      external-amf-ip: 10.201.0.100
+      external-amf-hostname: amf.mgmt
+EOF
+```
+
+Next, we create a Juju model to represent the Control Plane, using the cloud `control-plane-cluster`, which was created in the Bootstrap step.
+
+```console
+juju add-model control-plane control-plane-cluster
+```
+
+Deploy the control plane Juju bundle:
+```console
+juju deploy sdcore-control-plane --trust --channel=edge --overlay control-plane-overlay.yaml
+```
+
+Expose the Software as a Service offer for the AMF.
+
+```console
+juju offer control-plane.amf:fiveg-n2
+```
+
+Configure the NMS to be served as `control-plane.mgmt`:
+
+```console
+juju config traefik-k8s external_hostname=control-plane.mgmt
+```
+
+### Checkpoint
+
+You should be able to see the AMF External load balancer service in Kubernetes. Log into the control plane cluster and execute the following command:
+
+```console
+microk8s.kubectl get services -A | grep LoadBalancer
```

Comment:
We never mention adding user to `snap_microk8s` group in this host, so this command will fail.

Reply:
done